# ANALISIS PENDAPATAN - BELANJA APBN INDONESIA PERIODE 1990 - 2013

In [2]:
# import library
import pandas as pd
import math

# Analisis data_1 sebagai APBN Pendapatan

In [17]:
# read data 1 csv
data_1 = pd.read_csv('apbn-pendapatan-indonesia.csv', delimiter = ',', encoding = 'utf-8')
data_1

tahun sumber_pendapatan                    keterangan_sumber_pendapatan  \
0     1990       Bukan Pajak                          Surplus Bank Indonesia   
1     1990       Bukan Pajak                  SDA - Sumber Daya Alam Lainnya   
2     1990       Bukan Pajak                   Pendapatan Badan Layanan Umum   
3     1990       Bukan Pajak                                  SDA - Gas Alam   
4     1990       Bukan Pajak                                Bagian Laba BUMN   
5     1990       Bukan Pajak                                    PNBP lainnya   
6     1990       Bukan Pajak                               SDA - Minyak Bumi   
7     1990             Hibah                                           Hibah   
8     1990             Pajak                        Pajak Dalam Negeri - PPN   
9     1990             Pajak                      Pajak Dalam Negeri - BPHTB   
10    1990             Pajak                  Pajak Dalam Negeri - PPH Migas   
11    1990             Pajak                        Pajak Dalam Negeri - PBB   
12    1990             Pajak  Pajak Perdagangan Internasional - Pajak Ekspor   
13    1990             Pajak                      Pajak Dalam Negeri - Cukai   
14    1990             Pajak     Pajak Perdagangan Internasional - Bea Masuk   
15    1990             Pajak              Pajak Dalam Negeri - Pajak Lainnya   
16    1990             Pajak              Pajak Dalam Negeri - PPH Non Migas   
17    1991       Bukan Pajak                   Pendapatan Badan Layanan Umum   
18    1991       Bukan Pajak                          Surplus Bank Indonesia   
19    1991       Bukan Pajak                  SDA - Sumber Daya Alam Lainnya   
20    1991       Bukan Pajak                                    PNBP lainnya   
21    1991       Bukan Pajak                               SDA - Minyak Bumi   
22    1991       Bukan Pajak                                  SDA - Gas Alam   
23    1991       Bukan Pajak                                Bagian Laba BUMN   
24    1991             Hibah                                           Hibah   
25    1991             Pajak                      Pajak Dalam Negeri - Cukai   
26    1991             Pajak                  Pajak Dalam Negeri - PPH Migas   
27    1991             Pajak     Pajak Perdagangan Internasional - Bea Masuk   
28    1991             Pajak                        Pajak Dalam Negeri - PBB   
29    1991             Pajak                        Pajak Dalam Negeri - PPN   
..     ...               ...                                             ...   
378   2012       Bukan Pajak                          Surplus Bank Indonesia   
379   2012       Bukan Pajak                                Bagian Laba BUMN   
380   2012       Bukan Pajak                   Pendapatan Badan Layanan Umum   
381   2012             Hibah                                           Hibah   
382   2012             Pajak                        Pajak Dalam Negeri - PPN   
383   2012             Pajak                        Pajak Dalam Negeri - PBB   
384   2012             Pajak                      Pajak Dalam Negeri - BPHTB   
385   2012             Pajak     Pajak Perdagangan Internasional - Bea Masuk   
386   2012             Pajak                  Pajak Dalam Negeri - PPH Migas   
387   2012             Pajak              Pajak Dalam Negeri - PPH Non Migas   
388   2012             Pajak              Pajak Dalam Negeri - Pajak Lainnya   
389   2012             Pajak  Pajak Perdagangan Internasional - Pajak Ekspor   
390   2012             Pajak                      Pajak Dalam Negeri - Cukai   
391   2013       Bukan Pajak                                  SDA - Gas Alam   
392   2013       Bukan Pajak                  SDA - Sumber Daya Alam Lainnya   
393   2013       Bukan Pajak                                Bagian Laba BUMN   
394   2013       Bukan Pajak                               SDA - Minyak Bumi   
395   2013       Bukan Pajak                          Surplus Bank Indonesia   
396  

In [18]:
# check datatype
data_1.dtypes

tahun                             int64
sumber_pendapatan                object
keterangan_sumber_pendapatan     object
nilai                           float64
dtype: object

In [23]:
# look up columns
data_1.columns

Index([u'tahun', u'sumber_pendapatan', u'keterangan_sumber_pendapatan',
       u'nilai'],
      dtype='object')

In [26]:
# grouping data 1
data1_grouped = data_1.groupby(['tahun','sumber_pendapatan','keterangan_sumber_pendapatan'])

In [29]:
# count in data grouped
pendapatan = data1_grouped.sumber_pendapatan.count()
pendapatan

tahun  sumber_pendapatan  keterangan_sumber_pendapatan                  
1990   Bukan Pajak        Bagian Laba BUMN                                  1
                          PNBP lainnya                                      1
                          Pendapatan Badan Layanan Umum                     1
                          SDA - Gas Alam                                    1
                          SDA - Minyak Bumi                                 1
                          SDA - Sumber Daya Alam Lainnya                    1
                          Surplus Bank Indonesia                            1
       Hibah              Hibah                                             1
       Pajak              Pajak Dalam Negeri - BPHTB                        1
                          Pajak Dalam Negeri - Cukai                        1
                          Pajak Dalam Negeri - PBB                          1
                          Pajak Dalam Negeri - PPH Migas             

In [32]:
# stats describe data_1
total_pendapatan = data_1.nilai.sum()
rata_pendapatan = data_1.nilai.mean()
max_pendapatan = data_1.nilai.max()
min_pendapatan = data_1.nilai.min()

print(total_pendapatan)
print(rata_pendapatan)
print(max_pendapatan)
print(min_pendapatan)

10894639.3
26702.5473039
513509.0
0.0


In [89]:
# sum = data_1.groupby(['tahun','nilai']).sum()
# sum

In [38]:
data_1.loc[(data_1['nilai'] == max_pendapatan)]

tahun sumber_pendapatan        keterangan_sumber_pendapatan     nilai
399   2013             Pajak  Pajak Dalam Negeri - PPH Non Migas  513509.0

In [39]:
data_1.loc[(data_1['keterangan_sumber_pendapatan'] == 'Pajak Dalam Negeri - PPH Non Migas')]

tahun sumber_pendapatan        keterangan_sumber_pendapatan     nilai
16    1990             Pajak  Pajak Dalam Negeri - PPH Non Migas    6516.0
32    1991             Pajak  Pajak Dalam Negeri - PPH Non Migas    8021.0
45    1992             Pajak  Pajak Dalam Negeri - PPH Non Migas   10930.0
65    1993             Pajak  Pajak Dalam Negeri - PPH Non Migas   14849.0
80    1994             Pajak  Pajak Dalam Negeri - PPH Non Migas   18843.0
96    1995             Pajak  Pajak Dalam Negeri - PPH Non Migas   19239.0
114   1996             Pajak  Pajak Dalam Negeri - PPH Non Migas   23708.0
134   1997             Pajak  Pajak Dalam Negeri - PPH Non Migas   29118.0
152   1998             Pajak  Pajak Dalam Negeri - PPH Non Migas   25846.0
165   1999             Pajak  Pajak Dalam Negeri - PPH Non Migas    8021.0
179   2000             Pajak  Pajak Dalam Negeri - PPH Non Migas   40143.9
199   2001             Pajak  Pajak Dalam Negeri - PPH Non Migas   69696.0
214   2002             Pajak  Pajak Dalam Negeri - PPH Non Migas   87200.0
237   2003             Pajak  Pajak Dalam Negeri - PPH Non Migas  104305.0
246   2004             Pajak  Pajak Dalam Negeri - PPH Non Migas  112767.0
270   2005             Pajak  Pajak Dalam Negeri - PPH Non Migas  143017.0
281   2006             Pajak  Pajak Dalam Negeri - PPH Non Migas  175012.0
301   2007             Pajak  Pajak Dalam Negeri - PPH Non Migas  214481.0
315   2008             Pajak  Pajak Dalam Negeri - PPH Non Migas  251366.0
337   2009             Pajak  Pajak Dalam Negeri - PPH Non Migas  291176.0
353   2010             Pajak  Pajak Dalam Negeri - PPH Non Migas  306837.0
370   2011             Pajak  Pajak Dalam Negeri - PPH Non Migas  366746.0
387   2012             Pajak  Pajak Dalam Negeri - PPH Non Migas  445733.0
399   2013             Pajak  Pajak Dalam Negeri - PPH Non Migas  513509.0

In [88]:
# data_1.loc[(data_1['nilai'] == min_pendapatan)].groupby(['tahun','sumber_pendapatan','keterangan_sumber_pendapatan']).count()

In [81]:
# Visualize data_1
ax = data_1.groupby(data_1['keterangan_sumber_pendapatan'] == 'Pajak Dalam Negeri - PPH Non Migas')
ax = ax.plot(kind='bar', title ="Fluktuasi Pendapatan : PPH Non Migas", legend=True, fontsize=12)
ax.set_xlabel("tahun", fontsize=12)
ax.set_ylabel("nilai", fontsize=12)

AttributeError: 'Series' object has no attribute 'set_xlabel'

# Analisis data_2 sebagai APBN Belanja

In [21]:
# read data 2 csv
data_2 = pd.read_csv('apbn-belanja-indonesia.csv', delimiter = ',', encoding = 'utf-8')
data_2

tahun                                        pos_belanja  \
0     1990  Anggaran Belanja untuk Daerah / Transfer ke Da...   
1     1990  Anggaran Belanja untuk Daerah / Transfer ke Da...   
2     1990  Anggaran Belanja untuk Daerah / Transfer ke Da...   
3     1990  Anggaran Belanja untuk Daerah / Transfer ke Da...   
4     1990                           Belanja Pemerintah Pusat   
5     1990                           Belanja Pemerintah Pusat   
6     1990                           Belanja Pemerintah Pusat   
7     1990                           Belanja Pemerintah Pusat   
8     1990                           Belanja Pemerintah Pusat   
9     1990                           Belanja Pemerintah Pusat   
10    1990                           Belanja Pemerintah Pusat   
11    1990                           Belanja Pemerintah Pusat   
12    1990                           Belanja Pemerintah Pusat   
13    1990                           Belanja Pemerintah Pusat   
14    1990                           Belanja Pemerintah Pusat   
15    1990                           Belanja Pemerintah Pusat   
16    1990                                             Suspen   
17    1991  Anggaran Belanja untuk Daerah / Transfer ke Da...   
18    1991  Anggaran Belanja untuk Daerah / Transfer ke Da...   
19    1991  Anggaran Belanja untuk Daerah / Transfer ke Da...   
20    1991  Anggaran Belanja untuk Daerah / Transfer ke Da...   
21    1991                           Belanja Pemerintah Pusat   
22    1991                           Belanja Pemerintah Pusat   
23    1991                           Belanja Pemerintah Pusat   
24    1991                           Belanja Pemerintah Pusat   
25    1991                           Belanja Pemerintah Pusat   
26    1991                           Belanja Pemerintah Pusat   
27    1991                           Belanja Pemerintah Pusat   
28    1991                           Belanja Pemerintah Pusat   
29    1991                           Belanja Pemerintah Pusat   
..     ...                                                ...   
378   2012                           Belanja Pemerintah Pusat   
379   2012                           Belanja Pemerintah Pusat   
380   2012                           Belanja Pemerintah Pusat   
381   2012                           Belanja Pemerintah Pusat   
382   2012                           Belanja Pemerintah Pusat   
383   2012                           Belanja Pemerintah Pusat   
384   2012                           Belanja Pemerintah Pusat   
385   2012                           Belanja Pemerintah Pusat   
386   2012                           Belanja Pemerintah Pusat   
387   2012                           Belanja Pemerintah Pusat   
388   2012                           Belanja Pemerintah Pusat   
389   2012                           Belanja Pemerintah Pusat   
390   2012                                             Suspen   
391   2013  Anggaran Belanja untuk Daerah / Transfer ke Da...   
392   2013  Anggaran Belanja untuk Daerah / Transfer ke Da...   
393   2013  Anggaran Belanja untuk Daerah / Transfer ke Da...   
394   2013  Anggaran Belanja untuk Daerah / Transfer ke Da...   
395   2013                           Belanja Pemerintah Pusat   
396   2013                           Belanja Pemerintah Pusat   
397   2013                           Belanja Pemerintah Pusat   
398   2013                           Belanja Pemerintah Pusat   
399   2013                           Belanja Pemerintah Pusat   
400   2013                           Belanja Pemerintah Pusat   
401   2013                           Belanja Pemerintah Pusat   
402   2013                           Belanja Pemerintah Pusat   
403   2013                           Belanja Pemerintah Pusat   
404   2013                           Belanja Pemerintah Pusat   
405   2013                           Belanja Pemerintah Pusat   
406   2013                           Belanja Pemerintah Pusat   
407   2013                              

In [22]:
data_2.dtypes

tahun                       int64
pos_belanja                object
tipe_pengeluaran           object
keterangan_pengeluaran     object
nilai                     float64
dtype: object

In [24]:
# look up columns
data_2.columns

Index([u'tahun', u'pos_belanja', u'tipe_pengeluaran',
       u'keterangan_pengeluaran', u'nilai'],
      dtype='object')

In [52]:
# grouping data 2
data2_grouped = data_2.groupby(['tahun','pos_belanja','tipe_pengeluaran'])

In [53]:
pembelanjaan = data2_grouped.pos_belanja.count()
pembelanjaan

tahun  pos_belanja                                         tipe_pengeluaran                     
1990   Anggaran Belanja untuk Daerah / Transfer ke Daerah  Dana Otonomi Khusus dan Penyeimbangan     1
                                                           Dana Perimbangan                          3
       Belanja Pemerintah Pusat                            Pengeluaran Pembangunan                   2
                                                           Pengeluaran Rutin                        10
       Suspen                                              Suspen                                    1
1991   Anggaran Belanja untuk Daerah / Transfer ke Daerah  Dana Otonomi Khusus dan Penyeimbangan     1
                                                           Dana Perimbangan                          3
       Belanja Pemerintah Pusat                            Pengeluaran Pembangunan                   2
                                                           Pengeluaran Rutin   

In [54]:
total_belanja = data_2.nilai.sum()
rata_belanja = data_2.nilai.mean()
max_belanja = data_2.nilai.max()
min_belanja = data_2.nilai.min()

print(total_belanja)
print(rata_belanja)
print(max_belanja)
print(min_belanja)

8602820.7
23699.2305785
225534.0
-58721.0


In [87]:
# sum2 = data_2.groupby(['tahun','nilai']).sum()
# sum2

In [70]:
data_2.loc[(data_2['nilai'] == max_belanja)]

tahun                                        pos_belanja  \
359   2011  Anggaran Belanja untuk Daerah / Transfer ke Da...   

     tipe_pengeluaran keterangan_pengeluaran     nilai  
359  Dana Perimbangan      Dana Alokasi Umum  225534.0

In [86]:
# data_2.loc[(data_2['tipe_pengeluaran'] == 'Pengeluaran Rutin')]

In [75]:
data_2.loc[(data_2['tipe_pengeluaran'] == 'Dana Perimbangan') & (data_2['keterangan_pengeluaran'] == 'Dana Alokasi Umum')]

tahun                                        pos_belanja  \
2     1990  Anggaran Belanja untuk Daerah / Transfer ke Da...   
19    1991  Anggaran Belanja untuk Daerah / Transfer ke Da...   
36    1992  Anggaran Belanja untuk Daerah / Transfer ke Da...   
53    1993  Anggaran Belanja untuk Daerah / Transfer ke Da...   
70    1994  Anggaran Belanja untuk Daerah / Transfer ke Da...   
87    1995  Anggaran Belanja untuk Daerah / Transfer ke Da...   
104   1996  Anggaran Belanja untuk Daerah / Transfer ke Da...   
121   1997  Anggaran Belanja untuk Daerah / Transfer ke Da...   
138   1998  Anggaran Belanja untuk Daerah / Transfer ke Da...   
155   1999  Anggaran Belanja untuk Daerah / Transfer ke Da...   
172   2000  Anggaran Belanja untuk Daerah / Transfer ke Da...   
189   2001  Anggaran Belanja untuk Daerah / Transfer ke Da...   
206   2002  Anggaran Belanja untuk Daerah / Transfer ke Da...   
223   2003  Anggaran Belanja untuk Daerah / Transfer ke Da...   
240   2004  Anggaran Belanja untuk Daerah / Transfer ke Da...   
257   2005  Anggaran Belanja untuk Daerah / Transfer ke Da...   
274   2006  Anggaran Belanja untuk Daerah / Transfer ke Da...   
291   2007  Anggaran Belanja untuk Daerah / Transfer ke Da...   
308   2008  Anggaran Belanja untuk Daerah / Transfer ke Da...   
325   2009  Anggaran Belanja untuk Daerah / Transfer ke Da...   
342   2010  Anggaran Belanja untuk Daerah / Transfer ke Da...   
359   2011  Anggaran Belanja untuk Daerah / Transfer ke Da...   
376   2012  Anggaran Belanja untuk Daerah / Transfer ke Da...   
393   2013  Anggaran Belanja untuk Daerah / Transfer ke Da...   

     tipe_pengeluaran keterangan_pengeluaran     nilai  
2    Dana Perimbangan      Dana Alokasi Umum    6200.0  
19   Dana Perimbangan      Dana Alokasi Umum    7631.0  
36   Dana Perimbangan      Dana Alokasi Umum    9436.0  
53   Dana Perimbangan      Dana Alokasi Umum   11091.0  
70   Dana Perimbangan      Dana Alokasi Umum   12942.0  
87   Dana Perimbangan      Dana Alokasi Umum   13715.0  
104  Dana Perimbangan      Dana Alokasi Umum   15830.0  
121  Dana Perimbangan      Dana Alokasi Umum   18572.0  
138  Dana Perimbangan      Dana Alokasi Umum   22947.0  
155  Dana Perimbangan      Dana Alokasi Umum    7631.0  
172  Dana Perimbangan      Dana Alokasi Umum   28806.6  
189  Dana Perimbangan      Dana Alokasi Umum   60346.0  
206  Dana Perimbangan      Dana Alokasi Umum   69159.0  
223  Dana Perimbangan      Dana Alokasi Umum   76978.0  
240  Dana Perimbangan      Dana Alokasi Umum   82131.0  
257  Dana Perimbangan      Dana Alokasi Umum   88765.0  
274  Dana Perimbangan      Dana Alokasi Umum  145664.0  
291  Dana Perimbangan      Dana Alokasi Umum  164787.0  
308  Dana Perimbangan      Dana Alokasi Umum  179507.0  
325  Dana Perimbangan      Dana Alokasi Umum  186414.0  
342  Dana Perimbangan      Dana Alokasi Umum  203572.0  
359  Dana Perimbangan      Dana Alokasi Umum  225534.0  
376  Dana Perimbangan      Dana Alokasi Umum       NaN  
393  Dana Perimbangan      Dana Alokasi Umum       NaN

In [80]:
# Visualize data_2
ax = data_2.groupby(data_2['keterangan_pengeluaran'] == 'Dana Alokasi Umum')
ax = ax.plot(kind='bar', title ="Fluktuasi Pendapatan : Dana Alokasi Umum", legend=True, fontsize=12)
ax.set_xlabel("tahun", fontsize=12)
ax.set_ylabel("nilai", fontsize=12)

AttributeError: 'Series' object has no attribute 'set_xlabel'

# FINAL ANALYSIS

Berdasarkan analisis kedua jenis data APBN, yaitu APBN Pendapatan dan Pembelanjaan, periode 1990 - 2013, diperoleh beberapa hasil sebagai berikut :

1. Sumber pendapatan berbentuk pajak, bukan pajak, dan hibah, dimana masing-masing memiliki jenisnya masing-masing. Jenis bukan pajak adalah Bagian Laba BUMN, PNBP lainnya, Pendapatan Badan Layanan Umum, SDA (Gas Alam, Minyak Bumi, Sumber Daya Alam lainnya), dan surplus Bank Indonesia. Jenis hibah hanya dikategorikan satu yaitu hibah. Jenis pajak berupa pajak dalam negeri (BPHTB, Cukai, PBB, PPH Migas, PPH Non Migas, PPN, Pajak lainnya) dan pajak perdagangan internasional (bea masuk, pajak ekspor)
2. Jumlah pendapatan sepanjang tahun adalah 10894639,3 milyar rupiah.
3. Pendapatan tertinggi didapatkan di tahun 2013 pada sumber pendapatan pajak dalam negeri (PPH Non Migas) sebanyak 513509 milyar rupiah.
4. Jenis pendapatan PPH Non Migas selalu ada pada setiap tahun, tetapi mengalami penurunan pada rentang tahun 1998 - 1999 yang kemudian meningkat pesat di tahun 2000.
5. Pos belanja berbentuk anggaran untuk daerah, pemerintah pusat, dan suspen dimana masing-masing memiliki jenisnya masing-masing. Jenis anggaran untuk daerah adalah dana otonomi khusus dan penyeimbangan, dana perimbangan. Jenis suspen hanya dikategorikan satu yaitu suspen. Jenis anggaran untuk pemerintah pusat berupa pengeluaran pembangunan dan pengeluaran rutin, yang mana terdiri dari banyak tipe pengeluaran rutin.
6. Jumlah pendapatan sepanjang tahun adalah 8602820,7 milyar rupiah.
7. Pembelanjaan tertinggi dilakukan pada anggaran belanja untuk daerah pada jenis dana perimbangan sebagai dana alokasi umum.
8. Pada tahun 1999, dana alokasi umum ini mengalami penyusutan dibandingkan dengan tahun-tahun sebelumnya yang terus meningkat. Akan tetapi, tahun 2000, dana alokasi umum kembali meningkat hingga 2013.
9. Melihat total pendapatan dan pembelanjaan di tahun 1990 - 2013, selisih yang didapatkan tidak terlalu jauh, hanya mencapai kurang lebih 2 milyar rupiah. 
10. Akan tetapi, selisih yang diperoleh dapat lebih kecil dari itu karena terdapat beberapa data kosong (null) pada APBN belanja sehingga jumlah pembelanjaan diestimasikan dapat meningkat.